### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by LaskoA**
Please check it, thank you :)

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,90 @@
...
+        shop_list_each_name_each_store.append(sum_for_name_for_each_store)
+
+    for person in range(len(customers)):
+        budget = customers[person]["money"]
+        food = customers[person]['product_cart']
+        print(f"{customers[person]['name']} "
+              f"has {customers[person]['money']} dollars")
+        print(f"{customers[person]['name']}'s trip to the Outskirts Shop "
+              f"costs {shop_list_each_name_each_store[person][0]}")
+        print(f"{customers[person]['name']}'s trip to the Shop \'24/7\' "
```
Comment (Abnormaltype) [inline app/main.py:42]:
```suggestion
        print(f"{customers[person]['name']}'s trip to the Shop \'24/7\' "
```

create variable for {customers[person]['name'] and don't use \ in  \'24/7\'

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,90 @@
...
+            dist = ((my[0] - mall[0]) ** 2 + (my[1] - mall[1]) ** 2) ** 0.5
+            car_eats = customers[name]["car"]["fuel_consumption"]
+            fuel = fuel_price * dist * car_eats * 2 / 100
+            price += fuel
+            sum_for_name_for_each_store.append(round(price, 2))
+        shop_list_each_name_each_store.append(sum_for_name_for_each_store)
+
+    for person in range(len(customers)):
+        budget = customers[person]["money"]
+        food = customers[person]['product_cart']
```
Comment (Abnormaltype) [inline app/main.py:37]: use `""` instead od `''`

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,90 @@
...
+                  f"have enough money to make purchase in any shop")
+        else:
+            if min(shop_list_each_name_each_store[person]) == \
+                    shop_list_each_name_each_store[person][0]:
+                shop_id = 0
+                print(f"{customers[person]['name']} rides to Outskirts Shop")
+            elif min(shop_list_each_name_each_store[person]) == \
+                    shop_list_each_name_each_store[person][1]:
+                shop_id = 1
+                print(f"{customers[person]['name']} rides to Shop \'24/7\'")
```
Comment (Abnormaltype) [inline app/main.py:58]: the same here

File: app/main.py [hunk 4]
```diff
@@ -1,3 +1,90 @@
...
+            elif min(shop_list_each_name_each_store[person]) == \
+                    shop_list_each_name_each_store[person][2]:
+                shop_id = 2
+                print(f"{customers[person]['name']} rides to Central Shop")
+            print("")
+            print("Date: 11/03/2020 13:15:34")
```
Comment (Abnormaltype) [inline app/main.py:64]: it's not correct

File: app/main.py [hunk 5]
```diff
@@ -1,3 +1,90 @@
...
+                print(f"{customers[person]['name']} rides to Outskirts Shop")
+            elif min(shop_list_each_name_each_store[person]) == \
+                    shop_list_each_name_each_store[person][1]:
+                shop_id = 1
+                print(f"{customers[person]['name']} rides to Shop \'24/7\'")
+            elif min(shop_list_each_name_each_store[person]) == \
+                    shop_list_each_name_each_store[person][2]:
+                shop_id = 2
+                print(f"{customers[person]['name']} rides to Central Shop")
+            print("")
```
Comment (Abnormaltype) [inline app/main.py:63]: it's not correct, use` \n` in previous `print()`

File: app/main.py [hunk 6]
```diff
@@ -1,3 +1,90 @@
...
+            print(f"You have bought: \n"
+                  f"{food['milk']} milks for {milk_price} dollars\n"
+                  f"{food['bread']} breads for {bread_price} dollars\n"
+                  f"{food['butter']} butters for {butter_price} dollars")
+            print(f"Total cost is {total_cost} dollars")
+            print("See you again!")
+            print("")
+            print(f"{customers[person]['name']} rides home")
+            print(f"{customers[person]['name']} now has {money_left} dollars")
+            print("")
```
Comment (Abnormaltype) [inline app/main.py:90]: the same here

File: app/main.py [hunk 7]
```diff
@@ -1,3 +1,90 @@
...
+            fuel_cost = fuel_price * distance * car_drinks * 2 / 100
+            money_left = round(budget - total_cost - fuel_cost, 2)
+            print(f"Thanks, {customers[person]['name']}, for you purchase!")
+            print(f"You have bought: \n"
+                  f"{food['milk']} milks for {milk_price} dollars\n"
+                  f"{food['bread']} breads for {bread_price} dollars\n"
+                  f"{food['butter']} butters for {butter_price} dollars")
+            print(f"Total cost is {total_cost} dollars")
+            print("See you again!")
+            print("")
```
Comment (Abnormaltype) [inline app/main.py:87]: the same here

File: app/main.py [hunk 8]
```diff
@@ -1,3 +1,90 @@
...
+            y_coord_square = (home[1] - market[1]) ** 2
+            distance = (x_coord_square + y_coord_square) ** 0.5
+            car_drinks = customers[person]["car"]["fuel_consumption"]
+            fuel_cost = fuel_price * distance * car_drinks * 2 / 100
+            money_left = round(budget - total_cost - fuel_cost, 2)
+            print(f"Thanks, {customers[person]['name']}, for you purchase!")
+            print(f"You have bought: \n"
+                  f"{food['milk']} milks for {milk_price} dollars\n"
+                  f"{food['bread']} breads for {bread_price} dollars\n"
+                  f"{food['butter']} butters for {butter_price} dollars")
```
Comment (Abnormaltype) [inline app/main.py:84]: make loop please, DRY

File: app/main.py [hunk 9]
```diff
@@ -1,3 +1,90 @@
...
+                shop_id = 2
+                print(f"{customers[person]['name']} rides to Central Shop")
+            print("")
+            print("Date: 11/03/2020 13:15:34")
+            milk_price = customers[person]["product_cart"]["milk"] * \
+                shops[shop_id]["products"]["milk"]
+            bread_price = customers[person]["product_cart"]["bread"] * \
+                shops[shop_id]["products"]["bread"]
+            butter_price = customers[person]["product_cart"]["butter"] * \
+                shops[shop_id]["products"]["butter"]
```
Comment (Abnormaltype) [inline app/main.py:70]: make loop, DRY

**Comment by LaskoA** [comment]
Please check now, thank you)

File: app/main.py (hunk 2 above)
Comment (Abnormaltype) [reply app/main.py:39]: the same problem

File: app/main.py [hunk 10]
```diff
@@ -1,3 +1,93 @@
...
+            car_eats = customers[name]["car"]["fuel_consumption"]
+            fuel = fuel_price * dist * car_eats * 2 / 100
+            price += fuel
+            sum_for_name_for_each_store.append(round(price, 2))
+        shop_list_each_name_each_store.append(sum_for_name_for_each_store)
+
+    for person in range(len(customers)):
+        budget = customers[person]["money"]
+        food = customers[person]['product_cart']
+        name = customers[person]['name']
```
Comment (Abnormaltype) [inline app/main.py:40]: the same

**Review by Abnormaltype — CHANGES_REQUESTED** [review]

**Comment by LaskoA** [comment]
attempt #3....

**Review by Abnormaltype — CHANGES_REQUESTED** [review]
fix my suggestions

**Comment by LaskoA** [comment]
done)

**Review by Abnormaltype — APPROVED** [review]
good job!

**Pull request closed without merging**